tests/system/test_multigroups/test_multigroups.py

```diff
@@ -0,0 +1,259 @@
+    """Check that only the appropriate multi-groups are regenerated when a group file is created.
+
+    Check and store the modification time of all group and multigroup files. Create a new file inside
+    a group. Verify that only its 'merged.mg' and the files within the affected multigroup are regenerated.
+
+    Args:
+        target_group (str): Group in which to create the new file.
+    """
+    folders_to_check = set()
+    for agent_name, groups in agent_groups.items():
+        folders_to_check.update({os.path.join(shared_folder_path, group) for group in groups})
+        folders_to_check.add(os.path.join(mg_folder_path, calculate_mg_name(groups)))
+
+    # Check and store mtime of group files.
+    host_files = get_mtime(folders_to_check)
+
+    # Add new file to any group of agent1.
+    filename = f"new_file_{target_group}"
+    group_content = ''.join(random.choices(string.ascii_uppercase + string.digits, k=20))
+    group_path = os.path.join(shared_folder_path, target_group)
+    host_manager.modify_file_content(test_hosts[0], path=os.path.join(group_path, filename), content=group_content)
+
+    # Paths whose mtime should change after the new file was added.
+    expected_mtime_changed = [os.path.join(group_path, filename), os.path.join(group_path, 'merged.mg')]
+    expected_mtime_changed.extend([os.path.join(mg_folder_path, calculate_mg_name(groups)) for _, groups
+                                   in agent_groups.items() if target_group in groups])
+
+    sleep(time_to_sync)
+    new_host_files = get_mtime(folders_to_check)
+
+    for host, files in new_host_files.items():
+        for file, mtime in files.items():
+            for new_mtime_file in expected_mtime_changed:
+                if new_mtime_file in file:
+                    try:
+                        assert mtime != host_files[host][file], f"This file should have changed in {host}: {file}"
+                    except KeyError:
+                        assert os.path.basename(file) == filename
+                        assert group_content == host_manager.run_command(host, f"cat {file}"), \
+                            f"The new file content is not the expected in {file}"
+                    break
+            else:
+                assert mtime == host_files[host][file], f"This file changed its modification time in {host}: {file}"
+
+
+def test_multigroups_deleted(clean_files, agent_healthcheck, create_multigroups):
+    """Check that multigroups are removed when expected.
+
+    Unassign an agent from their groups or delete the groups. Check that the associated
+    multigroup disappears in both cases.
+    """
+    for agent_name, groups in agent_groups.items():
+        # Check that multigroups exists for each agent.
+        mg_path = os.path.join(mg_folder_path, calculate_mg_name(groups))
+        assert '' != host_manager.run_shell(test_hosts[0], f"ls {mg_path}"), f"{mg_path} should exist, but it does not."
```

Comment:
We have already checked if the path existed in the fixture. Also, we could calculate the multigroup folder name at the beginning so we don't calculate it twice.